yt/data_objects/construction_data_containers.py

```diff
@@ -899,6 +915,37 @@ def _fill_fields(self, fields):
             fi = self.ds._get_field_info(field)
             self[field] = self.ds.arr(dest, fi.units)
 
+    def _fill_sph_particles(self, fields):
+        # checks that we have the field and gets information
+        fields = [f for f in fields if f not in self.field_data]
+        if len(fields) == 0: return
+
+        ptype = self.ds._sph_ptype
+
+        for field in fields:
+            dest = np.zeros(self.ActiveDimensions, dtype="float64")
+
+            for chunk in self._data_source.chunks(fields, "io"):
+                px = chunk[(ptype,'particle_position_x')].in_units('cm')
+                py = chunk[(ptype,'particle_position_y')].in_units('cm')
+                pz = chunk[(ptype,'particle_position_z')].in_units('cm')
+                hsml = chunk[(ptype,'smoothing_length')].in_units('cm')*10.0
+                mass = chunk[(ptype,'particle_mass')].in_units('g')
+                dens = chunk[(ptype,'density')].in_units('g/cm**3')
```

Comment:
we could probably be doing this in code units. both here and in the image pixelizers. it doesn't really matter which unit system we choose as long as we stick to a single system, but code units is probably the most natural choice.